client-app/pages/catalog/catalog.vue

```diff
@@ -340,10 +339,7 @@ const itemsPerPage = ref(defaultPageSize);
 
 const stickyMobileHeaderAnchorIsVisible = useElementVisibility(stickyMobileHeaderAnchor, { direction: "top" });
 
-const viewModeQueryParam = useRouteQueryParam<"grid" | "list">("viewMode", {
-  defaultValue: "grid",
-  validator: (value) => (isMobile.value ? false : ["grid", "list"].includes(value)),
-});
+const viewMode = useLocalStorage<"grid" | "list">("viewMode", "grid");
```

Comment:
We have exactly same logic on the `search.vue` page. Need to sync in both places.